src/tox/config/__init__.py

```diff
@@ -1728,11 +1771,11 @@ def _replace(self, value, name=None, section_name=None, crossonly=False):
 class Replacer:
     RE_ITEM_REF = re.compile(
         r"""
-        (?<!\\)[{]
+        (?<![^\\]\\)[{]
         (?:(?P<sub_type>[^[:{}]+):)?    # optional sub_type for special rules
         (?P<substitution_value>(?:\[[^,{}]*\])?[^:,{}]*)  # substitution key
-        (?::(?P<default_value>[^{}]*))?   # default value
-        [}]
+        (?::(?P<default_value>([^{}]|\\{|\\})*))?   # default value
+        (?<!\\)[}]
```

Comment:
This should really be `(?<![^\\]\\)[}]` , but that introduces problems that I believe can only be solved if `\` is not permitted in section headings or in key names, or possibly only the last character of those strings.